# 1

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Используя математические операции, посчитать коэффициенты линейной регрессии, приняв за X заработную плату (то есть, zp - признак), а за y - значения скорингового балла (то есть, ks - целевая переменная). Произвести расчет как с использованием intercept, так и без.

In [1]:
import numpy as np

# количество наблюдений
n = 10

x = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110], dtype=np.float64)
y = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832], dtype=np.float64)

Для нахождения коэффициентов регрессии  𝑎  и  𝑏  воспользуемся приведенными выше формулами:

In [2]:
b = (np.mean(x * y) - np.mean(x) * np.mean(y)) / (np.mean(x**2) - np.mean(x) ** 2)
b

2.620538882402765

In [3]:
a = np.mean(y) - b * np.mean(x)
a

444.1773573243596

Итак, уравнение регрессии имеет вид (коэффициенты округлены до сотых):

𝑦=444.1+2.62⋅𝑥
С увеличением зарплаты на 1 доллар кредитный скоринг возрастет на 2.62 пункта.

Найдем коэффициент корреляции  𝑟  с помощью коэффициента  𝑏  и средних квадратического отклонения, посчитанного для массивов  𝑥  и  𝑦 :

In [6]:
r = b * np.std(x) / np.std(y)
r

0.8874900920739162

Найдем коэффициент детерминации  𝑅2 :

In [7]:
R2 = r**2
R2

0.7876386635293682

Это означает, что 78.7% вариации кредитного скоринга ( 𝑦 ) объясняется вариацией фактора  𝑥  — зарплатой.

С помощью этого уравнения регрессии посчитаем значения, предсказанные моделью значения цен на квартиры:

# 2

In [10]:
В каких случаях для вычисления доверительных интервалов и проверки статистических гипотез используется таблица значений функции Лапласа, а в каких - таблица критических точек распределения Стьюдента?

Object `Стьюдента` not found.


В каких случаях для вычисления доверительных интервалов и проверки статистических гипотез используется таблица значений функции Лапласа, а в каких - таблица критических точек распределения Стьюдента

Это зависит от того был ли порведен анализ генеральной совокупности.